expertsystem/data.py

```diff
@@ -164,16 +147,27 @@ def mass(self) -> float:
     def width(self) -> float:
         return self.__energy.imag
 
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, ComplexEnergy):
+            return self.complex_energy == other.complex_energy
+        raise NotImplementedError
 
-class ComplexEnergyState(HasComplexEnergy):
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}{self.complex_energy}"
```

Comment:
This would return something like `"ComplexEnergycomplex(2.5, 0.3)"`. So should be:
```python
return f"{self.__class__.__name__}{self.complex_energy}"
```
(in other cases in this module, omitting the parentheses was fine, because those were tuples).

Reply:
Good point. Somehow it creates `ComplexEnergy(2.5+j0.3)` though and everything is fine. Do you know how those brackets are automatically inserted?

Reply:
Oh interesting. Well that means the `__repr__` of a `complex` is just `(<real>+<imag>j)`.

Reply:
But how is `return f"{self.__class__.__name__}{self.complex_energy, self.state}"` for the `ComplexEnergyState` also correct then? Somehow this `{ ,,,, }` syntax is automatically creating a "tuple".

Reply:
No it's not a tuple, `{ ,,,, }`, just tenders the `__repr__` of a `complex`.